traffic_ops/traffic_ops_golang/cachegroup/cachegroups.go

```diff
@@ -436,13 +438,24 @@ func (cg *TOCacheGroup) Read() ([]interface{}, error, error, int) {
 	}
 	where, orderBy, pagination, queryValues, errs := dbhelpers.BuildWhereAndOrderByAndPagination(cg.ReqInfo.Params, queryParamsToQueryCols)
 	if len(errs) > 0 {
-		return nil, util.JoinErrs(errs), nil, http.StatusBadRequest
+		return nil, util.JoinErrs(errs), nil, http.StatusBadRequest, nil
+	}
+
+	runSecond, maxTime := ims.TryIfModifiedSinceQuery(cg.APIInfo().Tx, h, queryValues, selectMaxLastUpdatedQuery(where, orderBy, pagination))
+	if useIMS {
```

Comment:
So it looks like you addressed this in generic_crud.go but we should also skip the `TryIfModifiedSinceQuery` if IMS is disabled. I noticed this here and in topologies.go, so I imagine there might be other cases to still address in other files as well.